ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -598,7 +598,7 @@ export const PageProviderDisplayBehavior = [
             }
           }
           if (clear || this.items.length !== count) {
-            this.reset(count);
+            this.reset(response.pageSize < count ? response.pageSize : count);
```

Comment:
This fixes the issue with pagination, but won't if affect when using lazy scrolling? The items list is assumed to have the same size as the results count, and this will change right? I'm assuming people are relying on the `paginable` property, and if so, perhaps we could use it conditionally here to change the behavior?

Reply:
Ah yes, we could definitely go with a `this.paginable && response.pageSize < count` here, since the issue was only reported for paginated use cases.